flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCScanOptions.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.io.jdbc;
+
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Options for the JDBC scan.
+ */
+public class JDBCScanOptions implements Serializable {
+
+	private final String partitionColumnName;
+	private final long partitionLowerBound;
+	private final long partitionUpperBound;
+	private final int numPartitions;
+
+	private final int fetchSize;
+
+	private JDBCScanOptions(
+			String partitionColumnName,
+			long partitionLowerBound,
+			long partitionUpperBound,
+			int numPartitions,
+			int fetchSize) {
+		this.partitionColumnName = partitionColumnName;
+		this.partitionLowerBound = partitionLowerBound;
+		this.partitionUpperBound = partitionUpperBound;
+		this.numPartitions = numPartitions;
+
+		this.fetchSize = fetchSize;
+	}
+
+	public String getPartitionColumnName() {
```

Comment:
I would like to return an Optional instead of null. Returns null is always an error-prone design.
The following is the same.